File: Sahil/Microsoft/Day1/bulls-and-cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int x =0,y=0;
        int n = secret.length();
        for(int i=0;i<n;i++){
            if(secret[i]==guess[i]){
                x++;
            }
        }
        
        vector<int> mp1(10,0),mp2(10,0);
        for(int i=0;i<n;i++){
            mp1[secret[i] - '0']++;
        }
        for(int i=0;i<n;i++){
            mp2[guess[i] - '0']++;
        }
        
        for(int i=0;i<10;i++){
            y += min(mp1[i],mp2[i]);
        }
        
        return to_string(x) + 'A' + to_string(y-x) + 'B';
    }
};

File: Ravi/Microsoft/Day 1/BullsAndCows.java
class Solution {
    public String getHint(String secret, String guess) {
        
        int n = secret.length();
        
        StringBuilder sb = new StringBuilder("");
        
        int curr = 0;
        for(int i = 0; i < n; i++) {
            if(secret.charAt(i) == guess.charAt(i)) curr++;
        }
        
        sb.append(curr);
        sb.append('A');
        
        HashMap<Character, Integer> hm = new HashMap<>();
        for(int i = 0; i < n; i++) {
            hm.put(secret.charAt(i), hm.getOrDefault(secret.charAt(i), 0) + 1);
        }
        
        for(int i = 0; i < n; i++) {
            if(hm.containsKey(guess.charAt(i))) {
                curr--;
                hm.put(guess.charAt(i), hm.get(guess.charAt(i)) - 1);
                if(hm.get(guess.charAt(i)) == 0) hm.remove(guess.charAt(i));
            }
        }
        
        curr *= -1;
        sb.append(curr);
        sb.append('B');
        
        return sb.toString();
        
    }
}
